core/src/main/java/org/apache/iceberg/BaseContentScanTask.java

```diff
@@ -52,12 +53,24 @@ public F file() {
     return file;
   }
 
+  protected Schema schema() {
```

Comment:
exposed as `protected` so that `BaseFileScanTask` can use it to implement the `FileScanTask#schema()` method

Reply:
Little odd that we reverse engineer the schema from the string here, but seems like the most backwards compatible thing we can do here.